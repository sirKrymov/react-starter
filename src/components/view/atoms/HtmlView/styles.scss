.html-view {
  ul,
  ol {
    margin-bottom: 8px;

    &:last-child {
      margin-bottom: 0;
    }
  }

  li {
    margin-bottom: 8px;

    &:last-child {
      margin-bottom: 0;
    }
  }
}
