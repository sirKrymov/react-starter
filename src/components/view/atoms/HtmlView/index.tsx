// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import cn from 'classnames';

import './styles.scss';

interface IProps {
  className?: string;
  children: string;
}

export function HtmlView({ className, children }: IProps): ReactElement {
  const htmlViewClass = cn(
    {
      'html-view': true
    },
    className
  );

  return (
    <div
      className={htmlViewClass}
      dangerouslySetInnerHTML={{ __html: children }}
    />
  );
}
