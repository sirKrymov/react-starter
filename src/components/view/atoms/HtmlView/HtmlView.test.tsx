import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { HtmlView } from '../HtmlView';

describe('<HtmlView> atom component', () => {
  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    shallowComponent = shallow(<HtmlView>{htmlMock}</HtmlView>);
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});

const htmlMock = '<p>Mock test</p>';
