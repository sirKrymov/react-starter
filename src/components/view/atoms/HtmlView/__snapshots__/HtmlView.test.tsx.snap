// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlView> atom component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="html-view"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Mock test</p>",
    }
  }
/>
`;
